### PR TITLE
feat(unlock-app): Always show crypto button with amount needed to cover transaction

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

fixes https://github.com/unlock-protocol/unlock/issues/10005

CleanShot 2022-11-11 at 20 06 48@2x

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by searchableguy** [comment]
Is an explicit message required? I think it's self explanatory.

**Comment by julien51** [comment]
> Is an explicit message required? I think it's self explanatory.

Yes I think we need an explicit message, especially because there may be a misunderstanding between the need for Eth (or similar) to pay for gas and the need for ERC20.

File: unlock-app/src/components/interface/checkout/main/Payment.tsx [hunk 1]
```diff
@@ -29,24 +30,31 @@ interface Props {
   checkoutService: CheckoutService
 }
 
+interface AmountBadeProps {
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/main/Payment.tsx:33]:
```suggestion
interface AmountBadgeProps {
```

File: unlock-app/src/components/interface/checkout/main/Payment.tsx [hunk 2]
```diff
@@ -29,24 +30,31 @@ interface Props {
...
+  symbol: string
+  amount: string
+}
+
+const AmountBadge = ({ symbol, amount }: AmountBadeProps) => {
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/main/Payment.tsx:38]:
```suggestion
const AmountBadge = ({ symbol, amount }: AmountBadgeProps) => {
```

**Comment by searchableguy** [comment]
Updated to show message when user does not have gas fee to cover the transaction. 

For payment, we show the amount to be paid and balance. 

@julien51

**Review by julien51 — APPROVED** [review]
Very good!

**Pull request merged**